src/graph/entities/graph_entity.c

```diff
@@ -48,6 +49,14 @@ static void _GraphEntity_RemoveProperty(const GraphEntity *e, Attribute_ID attr_
 
 /* Add a new property to entity */
 SIValue *GraphEntity_AddProperty(GraphEntity *e, Attribute_ID attr_id, SIValue value) {
+	ASSERT(e);
+
+	// Emit an error and return if we're trying to add an invalid type.
+	if(!(SI_TYPE(value) & SI_PRIMITIVES)) {
+		QueryCtx_SetError("Property values can only be of primitive types.");
```

Comment:
Neo's error:
"Property values can only be of primitive types or arrays thereof"